corehq/apps/es/es_query.py

```diff
@@ -302,6 +302,31 @@ def remove_default_filter(self, default):
             query._default_filters.pop(default)
         return query
 
```

Comment:
@esoergel I think most of this PR is better for @czue  to review, but could I get your eyes on just the changes to `corehq/apps/es/*`? Also, shoutout for ESQuery—very easy to learn how to use and extend!